src/debug.sh

```diff
@@ -0,0 +1,527 @@
+
+      case "${options_values['TARGET']}" in
+        2) # LOCAL
+          cmd_manager 'SILENT' "sudo bash -c \"$disable_cmd\""
+          ;;
+        3 | 1) # REMOTE && VM
+          local remote="${remote_parameters['REMOTE_IP']}"
+          local port="${remote_parameters['REMOTE_PORT']}"
+          local user="${remote_parameters['REMOTE_USER']}"
+
+          cmd_remotely "$disable_cmd" 'SILENT' "$remote" "$port" "$user"
+          ;;
+      esac
+    fi
+  fi
+}
+
+function parser_debug_options()
+{
+  local options
+  local short_options
+  local long_options
+  local transition_variables
+
+  long_options='remote:,event:,cmd:,local,history,disable,list,follow,help'
+  short_options='f,e,c,h,d,l,k'
+
+  options="$(getopt -q --options "$short_options" \
+    --longoptions "$long_options" \
+    -- "$@")"
+
+  if [[ "$?" != 0 ]]; then
+    options_values['ERROR']="$(getopt --name 'debug' --options "$short_options" \
+      --longoptions "$long_options" \
+      -- "$@" 2>&1 > /dev/null)" # Get errors from stderr
+    return 22                    # EINVAL
+  fi
+
+  options_values['TARGET']=''
+  options_values['EVENT']=''
+  options_values['CMD']=''
+  options_values['HISTORY']=0
+  options_values['DISABLE']=0
+  options_values['LIST']=''
+  options_values['FOLLOW']=''
+
+  # Set default values
+  if [[ -n ${configurations[default_deploy_target]} ]]; then
+    transition_variables=${configurations[default_deploy_target]}
+    options_values['TARGET']=${deploy_target_opt[$transition_variables]}
+  else
+    options_values['TARGET']="$LOCAL_TARGET"
+  fi
+
+  populate_remote_info ''
+  if [[ "$?" == 22 ]]; then
+    options_values['ERROR']='Something is wrong in the remote option'
+    return 22 # EINVAL
+  fi
+
+  if [[ -n ${configurations[debug_event]} ]]; then
+    transition_variables=${configurations[debug_event]}
+    options_values['EVENT']=${deploy_target_opt[$transition_variables]}
+  fi
+
+  eval "set -- $options"
+  while [[ "$#" -gt 0 ]]; do
+    case "$1" in
+      --remote)
+        options_values['TARGET']="$REMOTE_TARGET"
+        populate_remote_info "$2"
+        if [[ "$?" == 22 ]]; then
+          options_values['ERROR']="$option"
+          return 22
+        fi
+        shift
+        ;;
+      --local)
+        options_values['TARGET']="$LOCAL_TARGET"
+        shift
+        ;;
+      --event | -e)
+        options_values['EVENT']+="$2"
+        shift 2
+        ;;
+      --cmd | -c)
+        options_values['CMD']+="$2"
+        shift 2
+        ;;
+      --history | -k)
+        options_values['HISTORY']=1
+        shift
+        ;;
+      --disable | -d)
+        options_values['DISABLE']=1
+        shift
+        ;;
+      --list | -l)
+        options_values['LIST']=1
+        shift
+        ;;
+      --follow | -f)
+        options_values['FOLLOW']=1
+        shift
+        ;;
+      --help | -h)
+        debug_help "$1"
+        exit
+        ;;
+      test_mode)
```

Comment:
This option is not on the list of options passed to `getopt` so its not going to be parsed correctly. Is this even needed? I noticed that `debug_manager()` is not invoked on the test suite.